pkg/common/aws/ec2.go

```diff
@@ -79,3 +85,49 @@ func (CcsAwsSession *ccsAwsSession) ReleaseElasticIPs(dryrun bool, sendSummary b
 
 	return nil
 }
+
+// TerminateEC2Instances finds EC2 instances with tag Name "osde2e*",
+// older than given duration, then terminates these EC2 instances.
+// Ignores EC2 instances with tag Name "osde2e-proxy*".
+func (CcsAwsSession *ccsAwsSession) TerminateEC2Instances(olderthan time.Duration, dryrun bool, sendSummary bool,
+	terminatedCounter *int, failedCounter *int, errorBuilder *strings.Builder,
+) error {
+	err := CcsAwsSession.GetAWSSessions()
+	if err != nil {
+		return err
+	}
+	result, err := CcsAwsSession.ec2.DescribeInstances(&ec2.DescribeInstancesInput{})
+	if err != nil {
+		return err
+	}
+	for _, reservation := range result.Reservations {
+		for _, instance := range reservation.Instances {
+			if time.Since(*instance.LaunchTime) > olderthan {
+				for _, tag := range instance.Tags {
+					if *tag.Key == "Name" && !strings.Contains(*tag.Value, osde2eproxysubstr) && strings.Contains(*tag.Value, osde2esubstr) {
```

Comment:
Why aren't we cleaning up _everything_ regardless of it containing this tag? This only cleans up instances that were created by osde2e but realistically it should delete everything regardless (someone comes in and creates 10 instances manually)

Reply:
I think it is fine to proceed with removing this part of condition. There is no reason to restrict something we know that we want to get rid of anyway for the sake of consistency

Reply:
I tried to follow approach similar to s3 buckets where only `osde2e` and `managed-velereo` are removed. But EC2 instances seem to be of only 2 types, `osde2e-proxy` and `osde2e`. Is it safe to assume we won't have any other types?

Reply:
I would default to deleting everything besides the proxy.